### ISSUE TITLE
Rust 1.19 regression, geogrid 0.3.0

### ISSUE THREAD

**Issue by brson** [rust-lang/rust#42458]
https://github.com/pelmers/geogrid

```
commit ab1fd34287e596cfb2e96de384c4c53311db430d
Author: Peter Elmers <peter.elmers@yahoo.com>
Date:   Wed Apr 5 18:00:20 2017 -0500

    Delete ocl suppression impl because it's slow
```

```
brian@ip-10-145-43-250:/mnt2/dev⟫ rustc +nightly -Vv
rustc 1.19.0-nightly (0418fa9d3 2017-06-04)
binary: rustc
commit-hash: 0418fa9d382a47d782cc1e195c14573be9c32095
commit-date: 2017-06-04
host: x86_64-unknown-linux-gnu
release: 1.19.0-nightly
LLVM version: 4.0

```

```
130 brian@ip-10-145-43-250:~/dev/geogrid⟫ cargo +nightly test
error: failed to parse manifest at `/mnt2/dev/geogrid/Cargo.toml`

Caused by:
  invalid type: string "true", expected a boolean for key `dependencies.ocl.optional`
```

**Comment by brson** [comment]
cc @pelmers

**Comment by Mark-Simulacrum** [comment]
Looks like this was already fixed on geogrid master in https://github.com/pelmers/geogrid/commit/c8a8b3dcd9016357772888444f6771428fab3043; but bisection shows introduction in d3abc80b3e601473e763b129acb636757ed96fc7 (#41830) which updated cargo.

Possibly related to https://github.com/rust-lang/cargo/pull/3964 (which updated Serde to 1.0.0 which removed some implicit conversions), but uncertain. This is the diff in Cargo: https://github.com/rust-lang/cargo/compare/fa7584c1495c2d9c04a6416f8e7b546abfa88a52...cf17c9f7118f544ec304ed6f50d92b3759487123.

**Comment by alexcrichton** [comment]
This was [removed from Serde](https://github.com/serde-rs/serde/pull/839) in the push to 1.0.0, but Cargo shipped in Rust 1.17.0 was the first Cargo to use Serde for deserialization (Serde 0.9). This was then also shipped in Rust 1.18.0 and the version of Cargo with Rust 1.19.0 (beta right now) uses Serde 1.0.0.

In that case this looks like it was a pretty small window of time that this bug was in Cargo and this has since been fixed upstream, so closing.
